### PR TITLE
Fail framing stage if length field is less than 0 

### LINKED ISSUE #22367: streams - Framing.lengthField() creates infinite loop if received length < 0
We discovered this in our production environment when an unknown external server hit our endpoint with an "unfortunate packet" and we saw massive CPU spikes. The following code shows the issue:
```
    implicit val sys = ActorSystem()
    implicit val mat = ActorMaterializer()
    implicit val bo = java.nio.ByteOrder.LITTLE_ENDIAN

    // A 4-byte message containing only an Int specifying the length of the payload
    // The issue shows itself if length in message is less than or equal
    // to -4 (if expected length field is length 4)
    val bs = ByteString.newBuilder.putInt(-4).result()

    val res: Future[Done] =
      Source
        .single(bs)
        .via(Flow[ByteString].via(Framing.lengthField(4, 0, 1000)))
        .runWith(Sink.foreach(println))

    // Future will not complete and standard out will have many ByteString() output
    import scala.concurrent.duration._
    Await.ready(res, 2.seconds)
```
I think the correct behaviour is that a `FramingException` should result from a packet where the indicated length is negative and a fix for that should be straightfoward. I'm happy to produce a PR for this if you want.

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #22367

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala [hunk 1]
```diff
@@ -266,6 +266,8 @@ object Framing {
           frameSize = parsedLength + minimumChunkSize
           if (frameSize > maximumFrameLength) {
             failStage(new FramingException(s"Maximum allowed frame size is $maximumFrameLength but decoded frame header reported size $frameSize"))
+          } else if (frameSize < 1) {
```
Comment (julianhowarth) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala]: Is it arguable that a length field of zero is valid? E.g. a client sending some form of keep-alive where there is no actual data, so length is set to zero. This isn't an issue for our use case but I wonder if others may be sending such packets in which case they will get failures when they upgrade. Instead, the current behaviour (of sending an empty `ByteString`) would be maintained if this specified `frameSize < 0`.

Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala]: Ok, good point. Will update to swallow 0 but fail on negative.

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala]: Yup, good point about the 0 there :)

**Review by ktoso — APPROVED** [review]
LGTM, thanks

**Comment by patriknw** [comment]
PLS BUILD

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
@johanandren please merge and backport after pr validation

**Pull request merged**